modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx

```diff
@@ -28,36 +29,147 @@ import {
 	december,
 	january,
 } from '../../../common/utils/dateFormatter';
+import {dataColumn} from './DataProductPerfomance';
+import {
+	BarChartPerformanceTypes,
+	DataChart,
+	MonthProperties,
+	Policy,
+	ProductListType,
+	SalesGoal,
+} from './ProductPerfomanceTypes';
 
-const TIME_PERIODS = ['YTD', '3 MO', '6 MO'];
-
-type Policy = {
-	boundDate: string;
-	productExternalReferenceCode: string;
-	productName: string;
-	termPremium: number;
+const PERIOD = {
+	SIX_MONTH: '1',
+	THREE_MONTH: '2',
+	YTD: '0',
 };
 
-type ProductListType = {
-	[keys: string]: ProductType;
+const BarChartPerformancee: BarChartPerformanceTypes = {
+	colors: [],
+	dataColumns: [],
+	groups: [''],
+	height: 338,
+	labelColumns: [],
+	showLegend: false,
+	showTooltip: true,
+	titleTotal: true,
+	totalSum: 0,
+	width: 700,
 };
 
-type ProductType = {
-	goalValue: number;
-	productName: string;
-	totalSales: number;
-};
+const TIME_PERIODS = [
+	{
+		label: '3 MO',
+		padding: 30,
+		value: '2',
+		width: 20,
+	},
+	{
+		label: '6 MO',
+		padding: 100,
+		value: '1',
+		width: 50,
+	},
+	{
+		label: 'YTD',
+		padding: 130,
+		value: '0',
+		width: 120,
+	},
+];
 
-type SalesGoal = {
-	finalReferenceDate?: string;
-	goalValue: number;
-	initialReferenceDate?: string;
-	productExternalReferenceCode: string;
+const colors: {[keys: string]: {}} = {
+	achieved: '#55C2FF',
+	exceeded: '#FFD76E',
+	goals: '#DCF1FD',
 };
 
+const date = new Date();
+const actualMonth = date.getMonth();
+
+const filterYearly = Object.values(dataColumn).filter(
+	(month: MonthProperties) => month.index <= actualMonth
+);
+
+const filterSix = Object.values(dataColumn).filter(
+	(month: MonthProperties) =>
+		month.index < actualMonth + 1 && month.index > actualMonth - 6
+);
+
+const filterThree = Object.values(dataColumn).filter(
+	(month: MonthProperties) =>
+		month.index < actualMonth + 1 && month.index > actualMonth - 3
+);
+
+const setLabelYearly = Object.values(dataColumn)
+	.filter((label: MonthProperties) => label.index <= actualMonth)
+	.map((label: MonthProperties) => label.label);
+
+const setLabelSix = Object.values(dataColumn)
+	.filter(
+		(label: MonthProperties) =>
+			label.index < actualMonth + 1 && label.index > actualMonth - 6
+	)
+	.map((label: MonthProperties) => label.label);
+
+const setLabelThree = Object.values(dataColumn)
+	.filter(
+		(label: MonthProperties) =>
+			label.index < actualMonth + 1 && label.index > actualMonth - 3
+	)
+	.map((label: MonthProperties) => label.label);
```

Comment:
Same here, you should try something like: 
```suggestion
function setLabelByPeriod (filteredPeriod) {
	const definedLabel = filteredPeriod.map((label: MonthProperties) => label.label);

	return definedLabel;
}
```

Reply:
Done!